dev/lint-python

```diff
@@ -17,8 +17,7 @@
 #
 # define test binaries + versions
 FLAKE8_BUILD="flake8"
-# TODO(SPARK-34943): minimum version should be 3.8+
-MINIMUM_FLAKE8="3.5.0"
+MINIMUM_FLAKE8="3.8.0"
```

Comment:
> The minimum flake8 version needs to be 3.8.0. Your current version is 3.6.0
Just for extra calrification, the version in Jenkins should be bumped up together before merging this PR.

Reply:
@HyukjinKwon Sorry, I have a problem. How to update the version in Jenkins?
I didn't find the other file that contains `flake8`

Reply:
That one has to be done by @shaneknapp so we would have to wait for his input.